bindings/java/iota-client-java/src/main/java/org/iota/apis/MiscellaneousApi.java

```diff
@@ -0,0 +1,131 @@
+import org.iota.types.secret.GenerateAddressesOptions;
+import org.iota.types.secret.GenerateBlockOptions;
+import org.iota.types.secret.SecretManager;
+
+public class MiscellaneousApi extends BaseApi {
+
+    public MiscellaneousApi(ClientConfig clientConfig) {
+        super(clientConfig);
+    }
+
+    public String[] generateAddresses(SecretManager secretManager, GenerateAddressesOptions generateAddressesOptions) throws ClientException {
+        JsonObject o = new JsonObject();
+        o.add("secretManager", secretManager.getJson());
+        o.add("options", generateAddressesOptions.getJson());
+
+        JsonArray responsePayload = (JsonArray) callBaseApi(new ClientCommand(ClientCommand.CommandType.CallClientMethod, "GenerateAddresses", o));
+
+        String[] addresses = new String[responsePayload.size()];
+        for (int i = 0; i < responsePayload.size(); i++) {
+            addresses[i] = responsePayload.get(i).getAsString();
+        }
+
+        return addresses;
+    }
+
+    public Block generateBlock(SecretManager secretManager, GenerateBlockOptions options) throws ClientException {
+        JsonObject o = new JsonObject();
+        o.add("secretManager", secretManager.getJson());
+        o.add("options", options.getJson());
+
+        JsonObject responsePayload = (JsonObject) callBaseApi(new ClientCommand(ClientCommand.CommandType.CallClientMethod, "GenerateBlock", o));
+
+        return new Block(responsePayload);
+    }
+
+
+    public Node getNode() throws ClientException {
+        JsonObject responsePayload = (JsonObject) callBaseApi(new ClientCommand(ClientCommand.CommandType.CallClientMethod, "GetNode"));
+        return new Node(responsePayload);
+    }
+
+    public JsonObject getNetworkInfo() throws ClientException {
+        JsonObject responsePayload = (JsonObject) callBaseApi(new ClientCommand(ClientCommand.CommandType.CallClientMethod, "GetNetworkInfo"));
+        return responsePayload;
+    }
+
+    public int getNetworkId() throws ClientException {
+        Integer responsePayload = callBaseApi(new ClientCommand(ClientCommand.CommandType.CallClientMethod, "GetNetworkId")).getAsInt();
+        return responsePayload;
+    }
+
+    public String getBech32Hrp() throws ClientException {
+        String responsePayload = callBaseApi(new ClientCommand(ClientCommand.CommandType.CallClientMethod, "GetBech32Hrp")).getAsString();
+        return responsePayload;
+    }
+
+    public float getMinPoWScore() throws ClientException {
+        Float responsePayload = callBaseApi(new ClientCommand(ClientCommand.CommandType.CallClientMethod, "GetMinPoWScore")).getAsFloat();
+        return responsePayload;
+    }
+
+    public int getTipsInterval() throws ClientException {
+        Integer responsePayload = callBaseApi(new ClientCommand(ClientCommand.CommandType.CallClientMethod, "GetTipsInterval")).getAsInt();
+        return responsePayload;
+    }
+
+    public boolean isLocalPow() throws ClientException {
```

Comment:
Why the different name? If it makes more sense to name things like this because they're local to the Client, then we should maybe also rename them in the Rust iota-client lib?

Reply:
In Java a good practice is to use the `is` prefix for boolean getters, so I thought it makes sense here to do (even if the Rust library doesn't) https://stackoverflow.com/a/5322666
In general for each binding it makes sense to follow its language specific best-practices for improved developer experience.. The "downside" is that the binding will differ a bit from the Rust library. What do you think?

Or are you talking about the "MiscellaneousApi" naming?

Reply:
> it makes sense to follow its language specific best-practices for improved developer experience
That's true, I didn't know about that this is a good practice for Java
On the other hand having it aligned for all languages seems to be easier to maintain and could also help to make things you see in one language work for the other one, for example some examples or guides
Not sure what we should go for

Reply:
thats true.. i guess we should have a discussion about this question.
And maybe until then, do like the Rust library

Reply:
changed it